src/utils.c

```diff
@@ -1927,4 +1940,8 @@ void vctrs_init_utils(SEXP ns) {
   // We assume the following in `union vctrs_dbl_indicator`
   VCTRS_ASSERT(sizeof(double) == sizeof(int64_t));
   VCTRS_ASSERT(sizeof(double) == 2 * sizeof(int));
+
+  // We assume the following in `vec_order()`
+  VCTRS_ASSERT(sizeof(int) == 4);
+  VCTRS_ASSERT(sizeof(double) == 8);
```

Comment:
I think we are essentially assuming the size of `int` here, since IEEE 754 encourages 64 bit doubles

```
VCTRS_ASSERT(sizeof(double) == 2 * sizeof(int));
```

I think R does too, since they define for `R_IsNaN()` (we do too, and it is the reason for the above assertion):

```
typedef union
{
    double value;
    unsigned int word[2];
} ieee_double;
```

https://github.com/wch/r-source/blob/abb550c99b3927e5fc03d12f1a8e7593fddc04d2/src/main/arithmetic.c#L85-L89

Reply:
> VCTRS_ASSERT(sizeof(double) == 2 * sizeof(int));

Sounds good.